primitives/application-crypto/Cargo.toml

```diff
@@ -18,22 +18,31 @@ targets = ["x86_64-unknown-linux-gnu"]
 sp-core = { version = "7.0.0", default-features = false, path = "../core" }
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
-serde = { version = "1.0.136", optional = true, features = ["derive"] }
+serde = { version = "1.0.136",  default-features = false, optional = true, features = ["derive"] }
 sp-std = { version = "5.0.0", default-features = false, path = "../std" }
 sp-io = { version = "7.0.0", default-features = false, path = "../io" }
 
 [features]
 default = [ "std" ]
 std = [
 	"full_crypto",
+	"serde_full",
 	"sp-core/std",
 	"codec/std",
 	"scale-info/std",
-	"serde",
+	"serde/std",
 	"sp-std/std",
 	"sp-io/std",
 ]
 
+# Serde support without relying on std features.
+serde_full = [
+	"full_crypto",
```

Comment:
```suggestion
```

Serde should not enable full_crypto

Reply:
Agreed, that's not nice. There's a problem with sp-runtime though. It implements `MultiSigner`

https://github.com/paritytech/substrate/blob/ed82c874b9cee5b7049730cf5983a02ebc9f89d5/primitives/runtime/src/lib.rs#L301-L310

But the serialization of `Public` is only available in the `full_crypto` feature, as it needs `to_ss58check()` :
https://github.com/paritytech/substrate/blob/ed82c874b9cee5b7049730cf5983a02ebc9f89d5/primitives/core/src/crypto.rs#L242-L243

Would it be alright to introduce the `full_crypto` feature in sp-runtime?